src/zenml/integrations/bentoml/steps/bentoml_deployer.py

```diff
@@ -0,0 +1,207 @@
+    pipeline_name = step_env.pipeline_name
+    run_id = step_env.pipeline_run_id
+    step_name = step_env.step_name
+
+    # fetch existing services with same pipeline name, step name and model name
+    existing_services = model_deployer.find_model_server(
+        pipeline_name=pipeline_name,
+        pipeline_step_name=step_name,
+        model_name=params.model_name,
+    )
+
+    # Return the apis endpoint of the defined service to use in the predict.
+    # This is a workaround to get the endpoints of the service defined as functions
+    # from the user code in the BentoML service.
+    def service_apis(bento_tag: str) -> List[str]:
+        # Add working dir in the bentoml load
+        service = bentoml.load(
+            bento_identifier=bento_tag,
+            working_dir=params.working_dir or str(repo_path),
+        )
+        apis = service.apis
+        apis_paths = list(apis.keys())
+        return apis_paths
+
+    # create a config for the new model service
+    predictor_cfg = BentoMLDeploymentConfig(
+        model_name=params.model_name,
+        bento=str(bento.tag),
+        model_uri=bento.info.labels.get("model_uri"),
+        bento_uri=bento.info.labels.get("bento_uri"),
+        apis=service_apis(str(bento.tag)),
+        workers=params.workers,
+        working_dir=params.working_dir or str(repo_path),
+        port=params.port,
+        pipeline_name=pipeline_name,
+        pipeline_run_id=run_id,
+        pipeline_step_name=step_name,
+        ssl_certfile=params.ssl_certfile,
+        ssl_keyfile=params.ssl_keyfile,
+        ssl_keyfile_password=params.ssl_keyfile_password,
+        ssl_version=params.ssl_version,
+        ssl_cert_reqs=params.ssl_cert_reqs,
+        ssl_ca_certs=params.ssl_ca_certs,
+        ssl_ciphers=params.ssl_ciphers,
+    )
+
+    # Creating a new service with inactive state and status by default
+    service = BentoMLDeploymentService(predictor_cfg)
+    if existing_services:
+        service = cast(BentoMLDeploymentService, existing_services[0])
+
+    if not deploy_decision and existing_services:
+        logger.info(
+            f"Skipping model deployment because the model quality does not "
+            f"meet the criteria. Reusing last model server deployed by step "
+            f"'{step_name}' and pipeline '{pipeline_name}' for model "
+            f"'{params.model_name}'..."
+        )
+        if not service.is_running:
+            service.start(timeout=params.timeout)
+        return service
+
+    # create a new model deployment and replace an old one if it exists
+    new_service = cast(
+        BentoMLDeploymentService,
+        model_deployer.deploy_model(
+            replace=True,
+            config=predictor_cfg,
+            timeout=params.timeout,
+        ),
+    )
+
+    logger.info(
+        f"BentoML deployment service started and reachable at:\n"
+        f"    {new_service.prediction_url}\n"
+    )
+
+    return new_service
+
+
+def bentoml_deployer_step(
```

Comment:
If this step is already deprecated before the initial release, can we simply remove it instead?

Reply:
Yes sure